### ISSUE TITLE
Installing with `chardet>=7.0.0` can break lexing due to differences in detected file encoding

### ISSUE THREAD

**Issue by peter-malcolm-bw** [sqlfluff/sqlfluff#7588]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

A new major version of [chardet](https://github.com/chardet/chardet?tab=readme-ov-file#whats-new-in-70) was recently released released which is **a ground up rewrite**.

<img width="658" height="304" alt="Image" src="https://github.com/user-attachments/assets/b781b706-07bc-4e48-a557-93b3dabe7c3a" />

The new package fails to detect some file encodings correctly that were previously detected correctly. Logs below.

```shell
== [path/to/file2.sql] FAIL
L:  13 | P:  79 |  LXR | Unable to lex characters: '\\x2b\\x31\\x...'
L:  79 | P:  10 |  PRS | Couldn't find closing bracket for opening bracket.
L:  95 | P:  60 |  LXR | Unable to lex characters: '귫Ắ\uf79d_col2...'
All Finished!
```

Is it worth pinning chardet to a last know good version until the new major version is stable 🙏 

Release history for chardet:
<img width="645" height="687" alt="Image" src="https://github.com/user-attachments/assets/8e0c96a9-e7d4-45e4-a319-727e760bc4fa" />

### Expected Behaviour

Detected file encodings remain consistent.

### Observed Behaviour

The new chardet version doesn't detect file encodings correctly in some cases.

### How to reproduce

Afraid I can't share the exact file contents here as this was for an internal repo. 

The key thing that caused issues was something like:

```sql
select sum(long_column_a+long_column_b) from table_c
```

Resolved by updating to (white space around the `+`):

```sql
select sum(long_column_a + long_column_b) from table_c
```

Confirming this was an encoding issue: 

```shell
uvx --from=chardet==7.0.0 python -c "import pathlib, chardet, sys; print(chardet.detect(pathlib.Path(sys.argv[1]).read_bytes())['encoding'])" path/to/file1.sql
utf-7

uvx --from=chardet==5.2.0 python -c "import pathlib, chardet, sys; print(chardet.detect(pathlib.Path(sys.argv[1]).read_bytes())['encoding'])" path/to/file1.sql
ascii
```

### Dialect

Snowflake

### Version

Affects multiple SQLfluff versions

### Configuration

Not relevant

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by keraion** [comment]
Would you mind testing your files against main? We applied a few fixes in #7575 but want to make sure this works for you.

**Comment by peter-malcolm-bw** [comment]
> Would you mind testing your files against main? We applied a few fixes in [#7575](https://github.com/sqlfluff/sqlfluff/pull/7575) but want to make sure this works for you.

Just tested. It works 🚀  

Not sure why I didn't find that PR when I looked 😄 

I did try running `pre-commit autoupdate` on the repo but that didn't find these changes as they are as-yet unreleased. Do you have any sense of when these changes might be released in a new sqlfluff version? 

Appreciate all your hard work!

I guess even when this is released this will still break the old versions of pre-commit hooks as the the only requirement currently is _some_ version of chardet? 

https://github.com/sqlfluff/sqlfluff/blob/main/pyproject.toml#L77

**Comment by keraion** [comment]
I don't have a timeline on the next release, but you should be able to add a chardet limit in pre-commit in the meantime.

```yaml
additional_dependencies:
  - "chardet<7.0.0"
```

**Comment by keraion** [comment]
Fixed by #7575.
